docs/sources/reference/commandline/cli.md

```diff
@@ -599,6 +606,28 @@ repository is used as Dockerfile. Note that you
 can specify an arbitrary Git repository by using the `git://` or `git@`
 schema.
 
+    $ sudo docker build -f Dockerfile.debug .
+
+This will use a file called `Dockerfile.debug` instead of `Dockerfile`
+during the build process. If a `Dockerfile` also exists it will be ignored.
+
+    $ sudo docker build -f dockerfiles/Dockerfile.debug -t myapp_debug .
+    $ sudo docker build -f dockerfiles/Dockerfile.prod  -t myapp_prod .
+
+The above commands will build the current build context twice, once
+using a debug version of a `Dockerfile` and once using a production
+version.
```

Comment:
its worth re-enforcing that the context is build from the location of the final `.`